pkg/descheduler/descheduler.go

```diff
@@ -69,7 +69,22 @@ func Run(ctx context.Context, rs *options.DeschedulerServer) error {
 		return err
 	}
 
-	return RunDeschedulerStrategies(ctx, rs, deschedulerPolicy, evictionPolicyGroupVersion)
+	runFn := func() error {
+		return RunDeschedulerStrategies(ctx, rs, deschedulerPolicy, evictionPolicyGroupVersion)
+	}
+
+	if rs.LeaderElection.LeaderElect && rs.DeschedulingInterval.Seconds() == 0 {
+		return fmt.Errorf("leaderElection must be used with deschedulingInterval")
+	}
+
+	if rs.LeaderElection.LeaderElect {
```

Comment:
`if rs.LeaderElection.LeaderElect && !rs.DryRun {`

No need to have the leader election on when the descheduler is run in the dry run mode. There's a use case for combination of the dry-run mode and the metrics enabled when two or more deschedulers (each with a different configuration) are run in parallel with metrics reported with goal to find a configuration which leads to an optional eviction.